### ISSUE TITLE
Latest docker desktop release (4.16.0) removes support for MacOS Catalina without any warning or notice in the release notes

### ISSUE THREAD

**Issue by SimoneGianni** [docker/for-mac#6671]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
Docker desktop should self-update and restart, or prevent the update if it's not compatible with the current system. 

### Actual behavior
Docker desktop self-updated and then refused to restart because the new 4.16.0 version does not support macOS Catalina 10.15 anymore.

I didn't get any warning from the self-update, nor it was reported anywhere, not even in the release notes, that Catalina support was being deprecated.

  - macOS Version: 10.15.7 (19H2)
  - Intel chip or Apple chip: Intel chip
  - Docker Desktop Version: 4.16.0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

Cannot run it. 

### Steps to reproduce the behavior

  1. Open Docker desktop 4.15 on MacOs Catalina 10.15.7 
  2. Click Settings
  3. Click Software Updates
  4. Click "Check for updates"
  5. See the message about the new version 4.16.0
  6. Notice it does not mention that it will not work on your system
  7. Notice the auto-updater is not preventing you to download it
  8. Click download
  9. Wait for the download
  10. Click restart
  11. It will not restart
  12. Try starting it manually
  13. Notice that MacOS is reporting that the new version of Docker 4.16 cannot be run on macOS Catalina

Workaround:
1. Download the previous version here https://desktop.docker.com/mac/main/amd64/93002/Docker.dmg
2. Install it manually
3. Do not update it

**Comment by marcelstoer** [comment]
I agree, a pretty stupid behavior. I almost expected this to happen when I realized Docker Desktop actually started downloading the update. And sure enough it failed to start after the update. I manually (re)installed 3.6 from https://docs.docker.com/desktop/previous-versions/3.x-mac/#docker-desktop-360.

**Comment by brunoc** [comment]
Problem confirmed (mbp intel / macos 10.15.7)

**Comment by actraiser** [comment]
Ran into the same issue - Docker asks to update from 4.15 to 4.16.1 and after installation it will not run on my iMac with MacOS Catalina. Reinstalled the old version from the release page.

Greets
-act

**Comment by CelsoSantos** [comment]
Also found this today. Not cool folks.. not cool!

**Comment by unostar** [comment]
Same here. Re-installed 4.15.0 from https://docs.docker.com/desktop/release-notes/#4150

**Comment by marcelstoer** [comment]
@unostar this report concerns macOS Catalina (10.15). How did you manage to install 4.15 on Catalina? Its release notes state

> Available for macOS 12.5 and above.

For 4.12 it says

> Minimum version to install or update Docker Desktop on macOS is now 10.15

No requires are documented for Docker Desktop 4.13 and 4.14. Hence, I was expecting the last version known to be working on Cataline be 4.12.

**Comment by unostar** [comment]
@marcelstoer 4.15 was my last working version on Catalina. I just re-installed from downloaded .dmg
Used to ignore system requirements, as they normally push to update, even if the app still works just fine.

**Comment by virtzilla** [comment]
> I agree, a pretty stupid behavior.
> 
> And I can't turn off the update check and the subsequent notifications. The setting is grayed out as I 'm not on a "Paid team plan".

This seems is brain dead behavior. 

Send customers right off the edge suggesting and proceeding with an upgrade that will send them down the path of a non-working docker install - and then on a path of manually finding, downloading and installing the version they were on...

And then continuing to push an upgrade to repeat the process.

They keep on forcing a dialog of "how do you rate this app".  In this case they'd get a zero if it was an option.

**Comment by xDmitriev** [comment]
Thank you Docker for such a great update, it stopped working on my Catalina without any notice. I switched to Lima instead  https://github.com/lima-vm/lima

**Comment by wolfgangbritzl** [comment]
Thanks for the help guys!
One of these moments I really hate open source... who does pushing this update without checking that could happen? Totally annoying.. shame on you docker...

**Comment by Jogan** [comment]
Hoping someone has a dmg available that works on 10.15.7. Looks like the dmg files were removed from the releases page.
